tests/conftest.py

```diff
@@ -70,273 +71,16 @@ def caplog(caplog: LogCaptureFixture) -> Generator[LogCaptureFixture, None, None
-            except FloatingPointError:
-                return float(np.finfo(float).max)
-
-        # It is quite important to start with a reasonable guess.
-        uncertainty_interval_midpoint = (lower_bound + upper_bound) / 2
-        # TODO: Further refine these concentration values. As long as we get convergence
-        # with one of them (for all the assertions we do), we're good -- but this specific
-        # list may not get us to convergence for future assertions we add.
-        for first_guess_concentration in [10_000, 1_000, 100, 10, 1, 0.5]:
-            optimization_result = scipy.optimize.minimize(
-                objective,
-                x0=[
-                    np.log(uncertainty_interval_midpoint * first_guess_concentration),
-                    np.log((1 - uncertainty_interval_midpoint) * first_guess_concentration),
-                ],
-            )
-            # Sometimes it warns that it *may* not have found a good solution,
-            # but the solution it found is very accurate.
-            if optimization_result.success or optimization_result.fun < 1e-05:
-                break
-
-        assert optimization_result.success or optimization_result.fun < 1e-05
-
-        result = np.exp(optimization_result.x)
-        assert len(result) == 2
-        return tuple(result)
-
-    def _quantile_squared_error(
-        self, dist: scipy.stats.rv_continuous, value: float, intended_quantile: float
-    ) -> float:
-        with np.errstate(under="ignore"):
-            actual_quantile = dist.cdf(value)
-
-        # Type hint
-        error: float
-        if 0 < actual_quantile < 1:
-            error = (
-                scipy.special.logit(actual_quantile) - scipy.special.logit(intended_quantile)
-            ) ** 2
-        else:
-            # In this case, we were so far off that the actual quantile can't even be
-            # precisely calculated.
-            # We return an arbitrarily large penalty to ensure this is never selected as the minimum.
-            error = np.finfo(float).max.astype(float)
-
-        return error
-
-    def save_diagnostic_output(self) -> None:
-        """
-        Save diagnostics for optional human inspection.
-        Can be useful to get more information about warnings, or to prioritize
-        areas to be more thorough in manual V&V.
-        """
-        output = pd.DataFrame(self.proportion_test_diagnostics)
-        output.to_csv(
-            Path(os.path.dirname(__file__))
-            / "v_and_v_output/proportion_test_diagnostics.csv",
-            index=False,
-        )
+@pytest.fixture(scope="session")
+def output_directory() -> Path:
+    v_v_path = Path(os.path.dirname(__file__)) / "v_and_v_output"
+    return v_v_path
 
 
 @pytest.fixture(scope="session")
-def fuzzy_checker() -> Generator[FuzzyChecker, None, None]:
+def fuzzy_checker(output_directory: Callable) -> Generator[FuzzyChecker, None, None]:
```

Comment:
does the fuzzy checker really take an output_directory _Callable_ instead of just the path? That seems weird